crates/criticalup-core/src/download_server_client.rs

```diff
@@ -124,9 +125,11 @@ impl DownloadServerClient {
             None
         };
 
+        let env_vars = state::EnvVars::read().await?;
+
         let header = self
             .state
-            .authentication_token(path_to_token_file)
+            .authentication_token(path_to_token_file, &env_vars)
```

Comment:
Noted elsewhere but this is the only usage of the token file path in our codebase, it creates some very confusing logic and perhaps we should refactor it out and have `EnvVars` intelligently detect if it's inside docker and look at `/run/secrets` when it's being built.